core/src/main/java/org/jivesoftware/LoginDialog.java

```diff
@@ -277,6 +284,22 @@ protected XMPPTCPConnectionConfiguration retrieveConnectionConfiguration() {
             builder.setProxyInfo( proxyInfo );
         }
 
+        if (securityMode != ConnectionConfiguration.SecurityMode.disabled && !useOldSSL) {
```

Comment:
(same as before - can you get rid of this code duplication?) Does this take into account that the user can also set another port than 5222? If that's done, we should not hard-code to 5222.